src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -77,13 +88,25 @@ public static void cacheDatabaseAsync() {
     thread.start();
   }
 
+  public static synchronized void verifyDatabaseCached() {
+    if (transitions == null) throw new IllegalStateException("Time Zone DB not loaded!");
+  }
+
   /**
    * Cache the database. This will take some time like several seconds.
    * If one `cacheDatabase` is running, other `cacheDatabase` will wait until caching is done.
    * If cache is exits, do not load cache again.
    */
+  public static void cacheDatabase(int maxYear) {
+    cacheDatabaseImpl(maxYear);
+  }
+
+  /**
+   * TO BE DELETED: added to not break spark-rapids build
+   * Will be removed after spark-rapids changes are committed
+   */
   public static void cacheDatabase() {
-    cacheDatabaseImpl();
+    cacheDatabaseImpl(9999);
```

Comment:
Wonder if this will impact any CI jobs which may have small memory space. I would suggest to use the default max year in https://github.com/NVIDIA/spark-rapids/pull/12502 to avoid any unnecessary hassles.